src/programs.rkt

```diff
@@ -222,6 +245,12 @@
      (list op
            (expand-associativity (cons op a))
            (expand-associativity b))]
+    [(cons (? (curryr member '(< <= > >= =)) op) a)
+     (define expanded (map expand-associativity a))
+     (if (> (length a) 2)
+         `(and ,@(for/list ([x expanded] [y (cdr expanded)])
+            (list op x y)))
+         (cons op expanded))]
```

Comment:
The comparison ops are expected to be variary, so we don't need to expand them here. Is that not so for the Posit ones? Probably better to write a wrapper in the posit bindings rather than expand them.

Reply:
`softposit-rkt` library has been updated to allow for variary comparison operators.